specification/workloadmonitor/resource-manager/Microsoft.WorkloadMonitor/preview/2020-01-13-preview/Microsoft.WorkloadMonitor.json

```diff
@@ -0,0 +1,805 @@
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "list of operations",
+            "schema": {
+              "$ref": "#/definitions/OperationList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/DefaultError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "OperationList_Get": {
+            "$ref": "./examples/OperationList_Get.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceNamespace}/{resourceType}/{resourceName}/providers/Microsoft.WorkloadMonitor/monitors": {
+      "get": {
+        "summary": "Get list of a monitors of a resource (with optional filter).",
+        "operationId": "Monitors_List",
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/resourceNamespace"
+          },
+          {
+            "$ref": "#/parameters/resourceType"
+          },
+          {
+            "$ref": "#/parameters/resourceName"
+          },
+          {
+            "$ref": "#/parameters/filter"
+          },
+          {
+            "$ref": "#/parameters/expand"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Search results matching criteria",
+            "schema": {
+              "$ref": "#/definitions/MonitorList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/DefaultError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get monitor list with default values": {
+            "$ref": "./examples/MonitorList_GetDefault.json"
+          },
+          "Get monitor list with filtered values": {
+            "$ref": "./examples/MonitorList_GetFiltered.json"
+          },
+          "Get monitor list with expanded values": {
+            "$ref": "./examples/MonitorList_GetExpanded.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceNamespace}/{resourceType}/{resourceName}/providers/Microsoft.WorkloadMonitor/monitors/{monitorId}": {
+      "get": {
+        "summary": "Get the current status of a monitor of a resource.",
+        "operationId": "MonitorInfo_Get",
```

Comment:
The OperationId is important in client generation, it uses the form <operationGroup>_<OperationName>, where OperationGroup is used to group together operations over the same resource.  In this case, the OperationGroup should likely be 'Monitors' to match the OperationGroup in the list operation above.

Reply:
I updated the operation ids to be: Operations_List, Monitors_ListMonitors, Monitors_GetMonitor, MonitorHistory_ListStateChanges, MonitorHistory_GetStateChange. Are these okay? I guess in this case the operation groups would be 'operations' 'monitors' and 'monitorhistory' -- is this okay or are these meant to match something in the url?

Reply:
Generally you don't want to repeat the operation group name in the operation name, so these should be:
Operations_list
Monitors_List

I would also argu that you want to have ListStateChanges and GetStateChange as operations organized as monitor operations, i.e.:

Monitors_ListStateChanges
Monitors_GetStateChange

When generating clients, this will organize these operations with the other monitor operations

if not, then 'MonitorHistory' or 'MonitorStateChanges' should be an operations group, and this would give you:
MonitorStateChanges_List
MonitorStateChanges_Get